### PR TITLE
bump(*): update etcd to 3.2.16 and grpc to 1.7.5

### PR DESCRIPTION
**Pull request by mfojtik**
Update of **etcd** level from 3.2.8 to **3.2.16** and gRPC to 1.7.5 (matching etcd version).

Fixes: https://github.com/openshift/origin/issues/18496

etcd:
- old origin: 3.2.8
- new origin: 3.2.16
- kube in 1.9: 3.1.11
- kube in 1.10: 3.2.13 (https://github.com/kubernetes/kubernetes/pull/57480)

grpc:
- old origin: 1.3.0
- new origin: 1.7.5
- kube in 1.9: 1.3.0
- kube in 1.10: 1.7.5 (https://github.com/kubernetes/kubernetes/pull/57480)

List of interesting changes or changes related to gRPC:

3.2.10: coreos/etcd@6d40628: update grpc, grpc-gateway (**1.4.2 -> 1.7.3**)
3.2.10: coreos/etcd@a8c84ff: clientv3: fix client balancer with gRPC v1.7
3.2.10: coreos/etcd@939337f: add max requests bytes, keepalive to server, blackhole methods to integration
3.2.10: coreos/etcd@8de0c04: Switch from boltdb v1.3.0 to coreos/bbolt v1.3.1-coreos.3 (<- concerning?)
3.2.11: coreos/etcd@5921b2c: log grpc stream send/recv errors in server-side
3.2.11: coreos/etcd@ff1f08c: upgrade grpc/grpc-go to v1.7.4
3.2.12: coreos/etcd@e82f055: clientv3: configure gRPC message limits in Config
3.2.12: coreos/etcd@c67e6d5: clientv3: call KV/Txn APIs with default gRPC call options
3.2.12: coreos/etcd@348b25f: clientv3: call other APIs with default gRPC call options
3.2.13: coreos/etcd@288ef7d: embed: fix gRPC server panic on GracefulStop
3.2.16: coreos/etcd@e08abbe: mvcc: restore unsynced watchers

@smarterclayton @deads2k @liggitt maybe too late in 3.9 cycle, but I don't see any huge risk change and this is a minor version bump that contain plenty of bug fixes.

This was clean bump, no build errors or panics during server start. There were 0 picks/carries on grpc or etcd.

@deads2k I wonder if we need to add grpc into our glide.yaml... If I don't and just bump the etcd, there are no changes in grpc, just etcd. I was worried that whatever higher level client we have that use grpc will use different version than etcd?

### CODE REVIEW

File: glide.lock [hunk 1]
```diff
@@ -1303,7 +1310,7 @@ imports:
 - name: gopkg.in/yaml.v2
   version: 53feefa2559fb8dfa8d81baad31be332c97d6c77
 - name: k8s.io/api
-  version: 78d52114c359511e7d0d5af4fc50148b5392d7b4
+  version: 30a449993d4d38589fa79362431bd9b04644cab5
```
Comment (mfojtik) [inline glide.lock]: @deads2k this is probably OK, reflecting the branches that are now synced by publishing bot.

File: glide.yaml [hunk 2]
```diff
@@ -93,7 +93,11 @@ import:
   version: ded73eae5db7e7a0ef6f55aace87a2873c5d2b74
 # master
 - package: github.com/coreos/etcd
-  version: v3.2.8
+  version: v3.2.16
+- package: google.golang.org/grpc
```
Comment (mfojtik) [inline glide.yaml]: this is pinning the version of grpc to the version used by etcd 3.2.16

Comment (soltysh) [reply glide.yaml]: Shouldn't that be discovered by glide?

Comment (mfojtik) [reply glide.yaml]: surprisingly not... when I just bumped etcd, these were unchanged and also etcd/vendor dir was unchanged which is pretty weird.

Comment (mfojtik) [reply glide.yaml]: What is strange is that the HEAD version of grpc is 1.3.0, but the etcd 3.2.8 version of grpc is 1.2.1 (from https://github.com/coreos/etcd/blob/v3.2.8/glide.yaml#L99). That means that in HEAD we are ahead of what our level of etcd was using.

Comment (liggitt) [reply glide.yaml]: > when I just bumped etcd, these were unchanged (on disk).

did you unpin these first? if not, pinned means pinned.

Comment (mfojtik) [reply glide.yaml]: @liggitt i don't see how they are currently pinned, we don't mention them in glide.yaml

**Comment by mfojtik** [comment]
For the unit test, there is no failure reported in log but it is reported as failed because of build failed? Looks like https://github.com/openshift/origin/issues/18497

**Comment by mfojtik** [comment]
hrm. ok the unit tests are maybe not flaking bug failing to build but I haven't found the reason yet.,

**Comment by stevekuznetsov** [comment]
@mfojtik if that is 1000% reproducible on test unit let's debug today

**Comment by mfojtik** [comment]
@stevekuznetsov seems like this PR didn't have a single green unit test run...

**Comment by mfojtik** [comment]
@stevekuznetsov ok, I found a panic in `vendor/k8s.io/kubernetes/pkg/master` :-)

**Comment by liggitt** [comment]
might need to pull in kube changes from a similar bump:

https://github.com/kubernetes/kubernetes/pull/57480/commits

**Comment by liggitt** [comment]
does updating grpc have implications for containerd, cadvisor, or kubelet APIs?

**Comment by mfojtik** [comment]
> might need to pull in kube changes from a similar bump:
> https://github.com/kubernetes/kubernetes/pull/57480/commits

As strange as it is, we seems to have some of these changes already incorporated...

**Comment by mfojtik** [comment]
I added bump(go-openapi/spec): 1de3e0542de65ad8d75452a595886fdd0befb363 as it claims to cut  ~25s from the test run. Also increased the unit test timeout to see if this will pass.

**Comment by mfojtik** [comment]
> does updating grpc have implications for containerd, cadvisor, or kubelet APIs?

@liggitt it does (from description of upstream bump):

Upgrade to containerd v1.0.0? Currently kubernetes depends on containerd v1.0.0-beta.2-159-g27d450a0 which depends on grpc v1.3.0, but containerd v1.0.0 depends on grpc 1.7.2. Not needed. The containerd grpc upgrade required no code changes.

I think we already are on containerd@1.0.2 so that should be no-op.

**Comment by liggitt** [comment]
> github.com/openshift/origin/pkg/cmd/server/kubernetes/master TestNewMasterLeasesHasCorrectTTL 0.12s
>
> go run hack/e2e.go -v -test --test_args='--ginkgo.focus=github\.com\/openshift\/origin\/pkg\/cmd\/server\/kubernetes\/master\sTestNewMasterLeasesHasCorrectTTL$'
>
> utils.go:293: Failed to start etcd server error=failed to initialize the etcd server: auth: invalid auth options

**Comment by liggitt** [comment]
> @liggitt are you ok with living with increased timeout till the upstream issue is fixed?

no, we need to know why this change destroyed our package test time:

before this PR:
```
ok  	github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/master	9.646s
```

with this PR:
```
ok  	github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/master	456.762s	
```

**Comment by liggitt** [comment]
fast
--- PASS: TestWriteClientCAs (0.00s)
--- PASS: TestReconcileEndpoints (0.02s)
--- PASS: TestCreateOrUpdateMasterService (0.00s)
--- PASS: TestGroupVersions (0.00s)
--- PASS: TestTypeTags (0.10s)
--- PASS: TestLegacyRestStorageStrategies (1.68s)
--- PASS: TestCertificatesRestStorageStrategies (1.78s)
--- PASS: TestVersion (2.32s)
--- PASS: TestGetNodeAddresses (0.00s)
--- PASS: TestAPIVersionOfDiscoveryEndpoints (1.52s)
--- PASS: TestNoAlphaVersionsEnabledByDefault (0.00s)

slow
--- PASS: TestWriteClientCAs (0.00s)
--- PASS: TestReconcileEndpoints (0.02s)
--- PASS: TestCreateOrUpdateMasterService (0.01s)
--- PASS: TestGroupVersions (0.00s)
--- PASS: TestTypeTags (0.23s)
--- PASS: TestLegacyRestStorageStrategies (4.51s)
--- PASS: TestCertificatesRestStorageStrategies (37.12s)
--- PASS: TestVersion (110.88s)
--- PASS: TestGetNodeAddresses (0.01s)
--- PASS: TestAPIVersionOfDiscoveryEndpoints (299.09s)
--- PASS: TestNoAlphaVersionsEnabledByDefault (0.00s)

**Comment by mfojtik** [comment]
@liggitt TestAPIVersionOfDiscoveryEndpoints ?

**Comment by mfojtik** [comment]
Moved to https://github.com/openshift/origin/pull/18731

**Pull request closed without merging**